File: docs/PadroesDeProjeto/src/Visitor/DiscountA.java
public class DiscountA implements Visitor {
    @Override
    public void visit(Food food) {
        System.out.println("Aplicando 10% de desconto na comida: " + food.getName());
        double discountedPrice = food.getPrice() * 0.9;
        System.out.println("Preço com desconto: " + discountedPrice);
    }

    @Override
    public void visit(Electronics electronics) {
        System.out.println("Aplicando 5% de desconto nos eletronicos: " + electronics.getName());
        double discountedPrice = electronics.getPrice() * 0.95;
        System.out.println("Preço com desconto: " + discountedPrice);
    }
}

File: docs/ArquiteturaReutilizacao/Observer-Visitor-Bridge/mymarket/src/visitors/DiscountA.js
import Visitor from './VIsitor1';

class DiscountA extends Visitor {
    visitarProduto(produto) {
        console.log(`Aplicando 10% de desconto em ${produto.nome}`);
        const precoComDesconto = produto.preco * 0.9;
        console.log(`Preço com desconto: R$${precoComDesconto}`);
    }
}

export default DiscountA;
